docs-go/get-started-go.md

```diff
@@ -0,0 +1,148 @@
+# Get started with Genkit using Go
+
+To get started with Firebase Genkit, install the Genkit CLI and run
+`genkit init` in a Go project. The rest of this page shows you how.
+
+## Requirements
+
+- Go 1.22 or later
+- Node.js 20 or later (for the Genkit CLI and UI)
+
+## Procedure
+
+1.  Install the Genkit CLI by running the following command:
+
+    ```posix-terminal
+    npm i -g genkit
+    ```
+
+1.  Create a new project directory:
+
+    ```posix-terminal
+    mkdir genkit-intro && cd genkit-intro
+    ```
+
+1.  Initialize a Genkit project:
+
+    ```posix-terminal
+    genkit init
+    ```
+
+    1. Select `Go` as the runtime environment.
+
+    1. Select your model:
+
+       - {Gemini (Google AI)}
+
+         The simplest way to get started is with Google AI Gemini API. Make sure
+         it's
+         [available in your region](https://ai.google.dev/available_regions).
+
+         [Generate an API key](https://aistudio.google.com/app/apikey) for the
+         Gemini API using Google AI Studio. Then, set the `GOOGLE_GENAI_API_KEY`
+         environment variable to your key:
+
+         ```posix-terminal
+         export GOOGLE_GENAI_API_KEY=<your API key>
+         ```
+
+       - {Gemini (Vertex AI)}
+
+         If the Google AI Gemini API is not available in your region, consider
+         using the Vertex AI API which also offers Gemini and other models. You
+         will need to have a billing-enabled Google Cloud project, enable AI
+         Platform API, and set some additional environment variables:
+
+         ```posix-terminal
+         gcloud services enable aiplatform.googleapis.com
+
+         export GCLOUD_PROJECT=<your project ID>
+
+         export GCLOUD_LOCATION=us-central1
+         ```
+
+         See https://cloud.google.com/vertex-ai/generative-ai/pricing for Vertex AI pricing.
+
+    1. Specify anything for the module name. For example: `example/genkit-intro`
+
+    1. Choose default answers to the rest of the questions, which will
+       initialize your project folder with some sample code.
+
+    The `genkit init` command creates a sample source file, `main.go`. This file demonstrates how you configure Genkit for your project, configure the plugins you want to load and export your AI flows and other resources you've defined. The sample file contains a config that loads a plugin to support the model provider you chose earlier. It also contains a single flow, `menuSuggestionFlow`, that prompts an LLM to suggest an item for a restaurant with a given theme.
+
+    ```go
+    err := googleai.Init(context.Background(), "")
```

Comment:
Use %include and weave:
- put code into go/internal/doc-snippets/get_started.go
- replace with %include lines
- rename this file get-started.src  (I don't think you need "-go" in the filename because it's in the dir name)
- add it to the Makefile